### ISSUE TITLE
Using SwinjectStoryboard with Swinject by adding the projects to an app project/workspace

### ISSUE THREAD

**Issue by thalmicMark** [Swinject/Swinject#184]
I'm having some trouble building a project that uses both `Swinject` and `SwinjectStoryboard`. My initial approach was to include both projects as sub-projects in my project, and then link to the products of the appropriate targets. But, since `SwinjectStoryboard` depends on `Swinject` as well, I get a weird signing error when the second `Swinject.framework` is signed. 

![screen shot 2016-10-31 at 11 05 23 am](https://cloud.githubusercontent.com/assets/5032263/19859011/190b0900-9f5a-11e6-97a4-0dbe1606fbd9.png)

The only way I've gotten it to work is to *just* include `SwinjectStoryboard`, and link to the `Swinject.framework` created in its subproject. 

How should I be organizing my project?

```
Xcode version 8.0 (8A218a)
Swinject (2.0.0-beta.2)
SwinjectStoryboard (1.0.0-beta.2)
```

I'm using Carthage to checkout the `Swinject` and `SwinjectStoryboard` dependencies, and adding the projects from `Carthage/Checkouts/` to my project as sub-projects. I would prefer a solution that uses subproject over using the Carthage-built framework.

**Comment by yoichitgy** [comment]
I'll investigate the issue. I understand your requirement because Swinject project itself is configured in that way without using prebuilt frameworks of Quick and Nimble by Carthage.

**Comment by thalmicMark** [comment]
Was this marked as an enhancement because there's a required change to be made to the Swinject project?

**Comment by yoichitgy** [comment]
Hi @thalmicMark. I marked this issue as an enhancement because I thought it was more like an enhancement rather than a bug. We'll put a milestone when we plan to resolve in an upcoming release. Usually one of committers takes time to investigate and resolve an issue, but a pull request by anyone is always welcome😃

**Comment by thalmicMark** [comment]
Cool thanks for the info!

**Comment by mpdifran** [comment]
I've looked into this issue and here's what I've found (all dependencies are added as sub-projects):

There's something weird with how Carthage resolves `SwinjectStoryboard`'s dependencies. I have a couple projects of my own (let's call them FrameworkA and FrameworkB) that link to `Swinject.framework` using the following CartFile:

    # FrameworkA and FrameworkB CartFile
    github "Swinject/Swinject" ~> 2.0.0

I've got the following CartFile in my main project:

    # Main project CartFile
    github "mpdifran/FrameworkA" ~> 2.2
    github "mpdifran/FrameworkB" ~> 1.0
    github "Swinject/SwinjectStoryboard" ~> 1.0.0

When running `carthage update`, Carthage is smart enough to checkout `Swinject` in the root `Carthage/Checkouts` directory, and add aliases inside `FrameworkA`'s and `FrameworkB`'s `Carthage/Checkouts` to the root one. 

This doesn't happen for `SwinjectStoryboard`. It gets its own `Swinject` project inside its local `Carthage/Checkouts`. I have no idea what makes `SwinjectStoryboard` different. I've tried modifying the CartFile on my fork as well to no avail. If anyone has more information as to what's going on here, I'd be glad to hear it. It's this duplicate `Swinject` project that's causing the issues here. 

Investigations using `Swinject` as a private dependency:

I tried updating all my frameworks, as well as `SwinjectStoryboard` to include `Swinject` in the `CartFile.private` file. This prevented the sharing of a common `Swinject` project in the `Carthage/Checkouts` directory. So I needed to add `Swinject` as a dependency in my main project's CartFile. Trying to build the project with this setup resulted in the same signing error though.

**Comment by mpdifran** [comment]
Ah I've figured out what the issue is, thanks to this issue on the Carthage project:https://github.com/Carthage/Carthage/issues/1700

It's because `SwinjectStoryboard` uses submodules. If you don't check out dependencies as submodules, the shared project is used.

Want me to make a PR to remove the submodules from `SwinjectStoryboard` and friends?

**Comment by yoichitgy** [comment]
Here similar thing might be talked, and I made some changes on SwinjectAutoregistration though I've not make a pull request yet.
https://github.com/Swinject/SwinjectAutoregistration/pull/11

I can't take time to see what you are going to do, so I'll reply in the next week again.

**Comment by jakubvano** [comment]
This have been resolved by [SwinjectStoryboard #44](https://github.com/Swinject/SwinjectStoryboard/pull/44), and related issue with SwinjectAutoregistration is tracked [here](https://github.com/Swinject/SwinjectAutoregistration/pull/11).

I suppose we can close this.
